### ISSUE TITLE
 XML.toJSONObject  should ignore the empty fileds 

### ISSUE THREAD

**Issue by smile18** [stleary/JSON-java#205]
currently when I convert my below XML to JSON using toJSONObject.. it's not ignoring the empty fileds ...
```
<parent>
<Child1>1</Child1><Child2>com</Child2><Child3>4423</Child3><Child4></Child4>
<NestedChild1><NChild1><NChildV1>170</NChildV1><NChildV2>Apt</NChildV2></NChild1></NestedChild1>
<NestedChild2></NestedChild2>
</parent>

```

 JSON I get is ....
`
{"parent":{"Child4":"","NestedChild1":{"NChild1":{"NChildV1":170,"NChildV2":"Apt"}},"NestedChild2":"","Child3":4423,"Child2":"com","Child1":1}}`

 I am expecting the json ..........
`
{"parent":{"NestedChild1":{"NChild1":{"NChildV1":170,"NChildV2":"Apt"}},"Child3":4423,"Child2":"com","Child1":1}}`

**Comment by johnjaylward** [comment]
Why would it ignore empty fields? That doesn't seem intuitive. IF it's defined in the XML, I would expect it to show in the JSON.

**Comment by stleary** [comment]
Thanks for bringing up this issue. There is no natural conversion between XML and JSON, and this example seems to be in line with the intended design. I don't think there is anything to be fixed here.

**Comment by smile18** [comment]
Sorry for the delay, I didn't look back here as I fixed this issue in my code with my own method. I raised this question because in case if nested filed is empty here in my case NestedChild2 my other service expects JSON like "NestedChild2":{}, it gives error if it is "NestedChild2":"" , so it's hard to decide in XML whether a filed is nested one or not so thought better to ignore empty field. Thanks for the response.
